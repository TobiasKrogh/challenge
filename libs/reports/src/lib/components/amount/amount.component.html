<ng-container *ngIf="amount">
  <ng-container *ngIf="suffix">{{ suffix }} </ng-container>
  {{ amount | formatAmount }}
  {{ 'CURRENCY' | translate }}
</ng-container>
