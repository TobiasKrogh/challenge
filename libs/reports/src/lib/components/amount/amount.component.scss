:host {
  display: block;
  font: var(--font-data--emphasizes);
}
