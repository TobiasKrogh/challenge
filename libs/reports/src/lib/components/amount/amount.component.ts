import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'reports-amount',
  templateUrl: './amount.component.html',
  styleUrls: ['./amount.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AmountComponent {
  @Input() amount: number | undefined;
  @Input() suffix: string | undefined;
}
