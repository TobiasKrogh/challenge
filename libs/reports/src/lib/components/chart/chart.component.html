<ng-container *ngIf="data$ | async as data">
  <div class="legend">
    <span
      *ngFor="let entry of data"
      [ngStyle]="{
        '--color': entry.color
      }"
      >{{ entry.label }}</span
    >
  </div>

  <div
    class="pie"
    [ngStyle]="{
      'background-image': pie$ | async
    }"
  >
    <span
      *ngFor="let entry of data"
      [ngStyle]="{
        transform: entry.labelTransform
      }"
    >
      {{ entry.percentage }}%
    </span>
  </div>
</ng-container>
