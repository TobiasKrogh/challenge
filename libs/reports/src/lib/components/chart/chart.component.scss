:host {
  display: block;
}

.legend {
  display: flex;
  justify-content: space-around;

  span {
    font: var(--font-chart-legend);
    padding-left: 1.75rem;
    position: relative;

    &::before {
      background-color: var(--color);
      border-radius: var(--border-radius--default);
      bottom: 0;
      content: '';
      left: 0;
      position: absolute;
      top: 0;
      width: 1rem;
    }
  }
}

.pie {
  border-radius: 50%;
  height: 16.875rem;
  margin: 2rem auto;
  position: relative;
  width: 16.875rem;

  &::after {
    background-color: var(--hex-color-white);
    border-radius: 50%;
    content: '';
    height: 8.75rem;
    left: 4.0625rem;
    position: absolute;
    top: 4.0625rem;
    width: 8.75rem;
  }

  span {
    color: var(--hex-color-white);
    display: block;
    font: var(--font-chart-label);
    height: 1.25rem;

    position: absolute;
    top: 50%;
    left: 50%;
    margin-left: -1.25rem;
    margin-top: -0.625rem;
    text-align: center;
    width: 2.5rem;
  }
}
