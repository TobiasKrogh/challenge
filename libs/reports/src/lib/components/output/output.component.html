<ng-container
  *ngIf="reportByGateway.length || reportByProject.length; else noData"
>
  <main class="data-display legend">
    <h3>
      <ng-container *ngIf="filters?.projectId as projectId; else allProjects">
        {{ projects?.[projectId]?.name }}
      </ng-container>
      <ng-template #allProjects>
        {{ 'REPORTS.ALL_PROJECTS' | translate }}
      </ng-template>
      |
      <ng-container *ngIf="filters?.gatewayId as gatewayId; else allGateways">
        {{ gateways?.[gatewayId]?.name }}
      </ng-container>
      <ng-template #allGateways>
        {{ 'REPORTS.ALL_GATEWAYS' | translate }}
      </ng-template>
    </h3>

    <section *ngFor="let report of reportByGateway">
      <ng-container
        *ngTemplateOutlet="
          table;
          context: { name: report.gateway.name, report }
        "
      ></ng-container>
    </section>

    <section *ngFor="let report of reportByProject">
      <ng-container
        *ngTemplateOutlet="
          table;
          context: { name: report.project.name, report }
        "
      ></ng-container>
    </section>
  </main>

  <aside *ngIf="showChart; else footer">
    <ng-container *ngIf="reportByGateway.length">
      <ng-container
        *ngTemplateOutlet="
          chartWithFooter;
          context: {
            sources: reportByGateway,
            suffix: 'REPORTS.TOTAL_GATEWAY' | translate
          }
        "
      ></ng-container>
    </ng-container>

    <ng-container *ngIf="reportByProject.length">
      <ng-container
        *ngTemplateOutlet="
          chartWithFooter;
          context: {
            sources: reportByProject,
            suffix: 'REPORTS.TOTAL_PROJECT' | translate
          }
        "
      ></ng-container>
    </ng-container>
  </aside>
</ng-container>

<ng-template #footer>
  <footer class="data-display legend">
    <reports-amount
      [amount]="total"
      [suffix]="'REPORTS.TOTAL' | translate"
    ></reports-amount>
  </footer>
</ng-template>

<ng-template #chartWithFooter let-sources="sources" let-suffix="suffix">
  <reports-chart [sources]="sources" [total]="total"></reports-chart>

  <footer class="data-display legend">
    <reports-amount [amount]="total" [suffix]="suffix"></reports-amount>
  </footer>
</ng-template>

<ng-template #table let-report="report" let-name="name">
  <div
    *ngIf="!isSingleGatewayAndProject"
    class="data-display legend"
    (click)="report.open = report.data.length > 1 ? !report.open : report.open"
  >
    <h4>
      {{ name }}
      <reports-amount
        [amount]="report.total"
        [suffix]="'REPORTS.TOTAL' | translate"
      ></reports-amount>
    </h4>
  </div>
  <table *ngIf="report.open">
    <colgroup>
      <col class="col-25" />
      <col *ngIf="showGateway" class="col-25" />
      <col class="col-25" />
      <col class="col-25" />
    </colgroup>
    <thead>
      <tr>
        <td>Date</td>
        <td *ngIf="showGateway" class="align--centered">Gateway</td>
        <td class="align--centered">Transaction ID</td>
        <td class="align--right">Amount</td>
      </tr>
    </thead>
    <tbody>
      <tr *ngFor="let entry of report.data; trackBy: trackByPaymentId">
        <td>{{ entry.created | formatDate }}</td>
        <td *ngIf="showGateway" class="align--centered">
          {{ gateways?.[entry.gatewayId]?.name }}
        </td>
        <td class="align--centered">{{ entry.paymentId }}</td>
        <td class="align--right">
          <reports-amount [amount]="entry.amount"></reports-amount>
        </td>
      </tr>
    </tbody>
  </table>
</ng-template>

<ng-template #noData>
  <div class="message-no-data">
    <shared-message
      [title]="'REPORTS.NO_DATA.TITLE' | translate"
      [text]="'REPORTS.NO_DATA.TEXT' | translate"
    ></shared-message>
  </div>
</ng-template>
