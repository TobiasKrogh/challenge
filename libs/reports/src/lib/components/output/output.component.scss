:host {
  --data-display-flex: 1;

  display: block;
  flex: 1;

  &.show-chart {
    --data-display-flex: 1 0 calc(55% - 1rem);
    --aside-flex: 0 0 calc(45% - 1rem);

    display: flex;
    flex-wrap: wrap;
    gap: 2rem;
  }
}

.data-display {
  flex: var(--data-display-flex);

  & & {
    align-items: center;
    background-color: var(--hex-color-white);
    cursor: pointer;
    display: flex;
    height: 4.5rem;
  }
}

main {
  min-width: 40rem;
}

main + footer {
  margin-top: 1.5rem;
}

aside {
  align-self: flex-start;
  flex: var(--aside-flex);
  position: sticky;
  top: 1.5rem;
}

h3 {
  font: var(--font-data--emphasizes);
  margin-bottom: 2rem;
}

section + section {
  margin-top: 0.875rem;
}

h4 {
  display: flex;
  flex: 1;
  font: var(--font-data--emphasizes);
  justify-content: space-between;
  margin-bottom: 0;
}

table {
  border-collapse: collapse;
  margin: 0.875rem 1rem 0 1rem;
  width: calc(100% - 2rem);

  .col-25 {
    width: 25%;
  }

  .align--centered {
    text-align: center;
  }

  .align--right {
    text-align: right;
  }

  tr {
    height: 2.25rem;

    td:first-child {
      padding-left: 0.25rem;
    }

    td:last-child {
      padding-right: 0.25rem;
    }
  }

  thead tr,
  tbody tr:nth-child(even) {
    background-color: var(--hex-color-white);
  }
}

.message-no-data {
  align-items: center;
  display: flex;
  justify-content: center;
  text-align: center;
}
