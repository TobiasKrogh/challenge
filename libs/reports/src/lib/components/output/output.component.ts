import {
  ChangeDetectionStrategy,
  Component,
  HostBinding,
  Input,
} from '@angular/core';
import {
  IFilters,
  IReport,
  TIndexedGateways,
  TIndexedProjects,
} from '@challenge/api';
import { IReportByGateway } from '../../model/report-by-gateway.interface';
import { IReportByProject } from '../../model/report-by-project.interface';

@Component({
  selector: 'reports-output',
  templateUrl: './output.component.html',
  styleUrls: ['./output.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class OutputComponent {
  @Input() public filters: IFilters | null | undefined;
  @Input() public gateways: TIndexedGateways | null = {};
  @Input() public projects: TIndexedProjects | null = {};

  public reportByGateway: IReportByGateway[] = [];
  public reportByProject: IReportByProject[] = [];

  protected total = 0;

  @Input() public set report(data: IReport[]) {
    this.reset();
    data = data.sort((a, b) => a.created.localeCompare(b.created));

    if (this.filters?.projectId && !this.filters.gatewayId) {
      this.reportByGateway = Object.values(this.gateways ?? {}).reduce(
        (acc: IReportByGateway[], gateway) => {
          const reduced = data.filter(
            ({ gatewayId }) => gatewayId === gateway.gatewayId
          );

          if (reduced.length) {
            acc.push({
              data: reduced,
              gateway,
              open: acc.length === 0,
              total: this.calculateTotal(reduced),
            });
          }

          return acc;
        },
        []
      );
    } else {
      this.reportByProject = Object.values(this.projects ?? {}).reduce(
        (acc: IReportByProject[], project) => {
          const reduced = data.filter(
            ({ projectId }) => projectId === project.projectId
          );

          if (reduced.length) {
            acc.push({
              data: reduced,
              open: acc.length === 0,
              project,
              total: this.calculateTotal(reduced),
            });
          }

          return acc;
        },
        []
      );
    }

    this.total = this.calculateTotal(data);
  }

  @HostBinding('class.show-chart')
  public get showChart(): boolean {
    return (
      (!!this.filters?.projectId && !this.filters?.gatewayId) ||
      (!this.filters?.projectId && !!this.filters?.gatewayId)
    );
  }

  public get showGateway(): boolean {
    return !this.filters?.gatewayId;
  }

  public get isSingleGatewayAndProject(): boolean {
    return !!this.filters?.projectId && !!this.filters?.gatewayId;
  }

  private reset(): void {
    this.reportByGateway = [];
    this.reportByProject = [];
  }

  public trackByPaymentId(_index: number, entry: IReport): string {
    return entry.paymentId;
  }

  private calculateTotal(data: IReport[]): number {
    return data.reduce((acc, entry) => acc + entry.amount, 0);
  }
}
