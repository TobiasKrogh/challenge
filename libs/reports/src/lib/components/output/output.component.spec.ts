import { NgIf } from '@angular/common';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TranslatePipe } from '@ngx-translate/core';
import { FormatDatePipe, MessageComponent } from '@challenge/shared';
import { MockComponents, MockPipes } from 'ng-mocks';
import {
  MOCK_GATEWAY,
  MOCK_PROJECT,
  MOCK_REPORT,
} from '../../../testing/data.mocks';
import { AmountComponent } from '../amount/amount.component';
import { ChartComponent } from '../chart/chart.component';

import { OutputComponent } from './output.component';

describe('OutputComponent', () => {
  let component: OutputComponent;
  let fixture: ComponentFixture<OutputComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NgIf, MessageComponent],
      declarations: [
        MockComponents(AmountComponent, ChartComponent),
        MockPipes(TranslatePipe),
        FormatDatePipe,
        OutputComponent,
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(OutputComponent);
    component = fixture.componentInstance;

    component.gateways = {
      ['gw-1']: { ...MOCK_GATEWAY, gatewayId: 'gw-1' },
      ['gw-2']: { ...MOCK_GATEWAY, gatewayId: 'gw-2' },
    };

    component.projects = {
      ['p-1']: { ...MOCK_PROJECT, projectId: 'p-1' },
      ['p-2']: { ...MOCK_PROJECT, projectId: 'p-2' },
    };
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot (no data provided)', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (multiple projects, multiple gateways)', () => {
      component.filters = {};
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-2',
          projectId: 'p-2',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (multiple projects, multiple gateways - nullified)', () => {
      component.filters = {};
      component.gateways = null;
      component.projects = null;
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-2',
          projectId: 'p-2',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (single project, multiple gateways)', () => {
      component.filters = {
        projectId: 'p-1',
      };
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-2',
          projectId: 'p-1',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (single project, multiple gateways - nullified)', () => {
      component.filters = {
        projectId: 'p-1',
      };
      component.gateways = null;
      component.projects = null;
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-2',
          projectId: 'p-1',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (multiple projects, single gateway)', () => {
      component.filters = {
        gatewayId: 'gw-1',
      };
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-1',
          projectId: 'p-2',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (single project, single gateway)', () => {
      component.filters = {
        projectId: 'p-1',
        gatewayId: 'gw-1',
      };
      component.report = [
        {
          ...MOCK_REPORT,
          paymentId: 'first',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
        {
          ...MOCK_REPORT,
          paymentId: 'second',
          gatewayId: 'gw-1',
          projectId: 'p-1',
        },
      ];
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });
  });
});
