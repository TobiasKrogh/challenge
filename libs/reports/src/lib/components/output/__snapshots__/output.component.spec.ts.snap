// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputComponent When rendering Then it matches snapshot (multiple projects, multiple gateways) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root2"
  >
    <main
      class="data-display legend"
    >
      <h3>
          
        
        
         | 
          
        
        
      </h3>
      
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        <table>
          <colgroup>
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
            
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
          </colgroup>
          <thead>
            <tr>
              <td>
                Date
              </td>
              <td
                class="align--centered"
              >
                Gateway
              </td>
              
              <td
                class="align--centered"
              >
                Transaction ID
              </td>
              <td
                class="align--right"
              >
                Amount
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                undefined/undefined/date
              </td>
              <td
                class="align--centered"
              >
                 name 
              </td>
              
              <td
                class="align--centered"
              >
                first
              </td>
              <td
                class="align--right"
              >
                <reports-amount />
              </td>
            </tr>
            
          </tbody>
        </table>
        
        
      </section>
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        
        
      </section>
      
    </main>
    <footer
      class="data-display legend"
    >
      <reports-amount />
    </footer>
    
    
    
    
    
    
    
  </div>,
}
`;

exports[`OutputComponent When rendering Then it matches snapshot (multiple projects, single gateway) 1`] = `
DebugElement {
  "nativeNode": <div
    class="show-chart"
    id="root4"
  >
    <main
      class="data-display legend"
    >
      <h3>
          
        
        
         | 
         name 
        
        
        
      </h3>
      
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        <table>
          <colgroup>
            <col
              class="col-25"
            />
            
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
          </colgroup>
          <thead>
            <tr>
              <td>
                Date
              </td>
              
              <td
                class="align--centered"
              >
                Transaction ID
              </td>
              <td
                class="align--right"
              >
                Amount
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                undefined/undefined/date
              </td>
              
              <td
                class="align--centered"
              >
                first
              </td>
              <td
                class="align--right"
              >
                <reports-amount />
              </td>
            </tr>
            
          </tbody>
        </table>
        
        
      </section>
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        
        
      </section>
      
    </main>
    <aside>
      
      <reports-chart />
      <footer
        class="data-display legend"
      >
        <reports-amount />
      </footer>
      
      
      
    </aside>
    
    
    
    
    
    
    
  </div>,
}
`;

exports[`OutputComponent When rendering Then it matches snapshot (no data provided) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root1"
  >
    <div
      class="message-no-data"
    >
      <shared-message
        class="title--with-subline"
      >
        
        
      </shared-message>
    </div>
    
    
    
    
    
  </div>,
}
`;

exports[`OutputComponent When rendering Then it matches snapshot (single project, multiple gateways) 1`] = `
DebugElement {
  "nativeNode": <div
    class="show-chart"
    id="root3"
  >
    <main
      class="data-display legend"
    >
      <h3>
         name 
        
        
        
         | 
          
        
        
      </h3>
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        <table>
          <colgroup>
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
            
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
          </colgroup>
          <thead>
            <tr>
              <td>
                Date
              </td>
              <td
                class="align--centered"
              >
                Gateway
              </td>
              
              <td
                class="align--centered"
              >
                Transaction ID
              </td>
              <td
                class="align--right"
              >
                Amount
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                undefined/undefined/date
              </td>
              <td
                class="align--centered"
              >
                 name 
              </td>
              
              <td
                class="align--centered"
              >
                first
              </td>
              <td
                class="align--right"
              >
                <reports-amount />
              </td>
            </tr>
            
          </tbody>
        </table>
        
        
      </section>
      <section>
        <div
          class="data-display legend"
        >
          <h4>
             name 
            <reports-amount />
          </h4>
        </div>
        
        
        
      </section>
      
      
    </main>
    <aside>
      <reports-chart />
      <footer
        class="data-display legend"
      >
        <reports-amount />
      </footer>
      
      
      
      
    </aside>
    
    
    
    
    
    
    
  </div>,
}
`;

exports[`OutputComponent When rendering Then it matches snapshot (single project, single gateway) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root5"
  >
    <main
      class="data-display legend"
    >
      <h3>
         name 
        
        
        
         | 
         name 
        
        
        
      </h3>
      
      <section>
        
        <table>
          <colgroup>
            <col
              class="col-25"
            />
            
            <col
              class="col-25"
            />
            <col
              class="col-25"
            />
          </colgroup>
          <thead>
            <tr>
              <td>
                Date
              </td>
              
              <td
                class="align--centered"
              >
                Transaction ID
              </td>
              <td
                class="align--right"
              >
                Amount
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                undefined/undefined/date
              </td>
              
              <td
                class="align--centered"
              >
                first
              </td>
              <td
                class="align--right"
              >
                <reports-amount />
              </td>
            </tr>
            <tr>
              <td>
                undefined/undefined/date
              </td>
              
              <td
                class="align--centered"
              >
                second
              </td>
              <td
                class="align--right"
              >
                <reports-amount />
              </td>
            </tr>
            
          </tbody>
        </table>
        
        
      </section>
      
    </main>
    <footer
      class="data-display legend"
    >
      <reports-amount />
    </footer>
    
    
    
    
    
    
    
  </div>,
}
`;
